packages/tools/viewer/src/viewer.ts

```diff
@@ -1411,9 +1608,127 @@ export class Viewer implements IDisposable {
+        );
+    }
+
+    /**
+     * Resets the viewer to its initial state based on the options passed in to the constructor.
+     * @param flags The flags that specify which parts of the viewer to reset. If no flags are provided, all parts will be reset.
      */
-    public resetCamera() {
-        this._camera.restoreState();
+    public reset(...flags: ResetFlag[]) {
+        this._reset(true, ...flags);
+    }
+
+    private _reset(interpolate: boolean, ...flags: ResetFlag[]) {
+        const handleLoadError = async (promise: Promise<unknown>) => {
+            try {
+                await promise;
+            } catch (error) {
+                if (!(error instanceof AbortError)) {
+                    Logger.Error(error);
+                }
+            }
+        };
+
+        if (flags.length === 0 || flags.includes("source")) {
+            handleLoadError(this._updateModel(this._options?.source));
+        }
+
+        if (flags.length === 0 || flags.includes("environment")) {
+            this._scene.clearColor = new Color4(...(this._options?.clearColor ?? DefaultViewerOptions.clearColor));
+            this.environmentConfig = {
+                intensity: this._options?.environmentConfig?.intensity ?? DefaultViewerOptions.environmentConfig.intensity,
+                blur: this._options?.environmentConfig?.blur ?? DefaultViewerOptions.environmentConfig.blur,
+                rotation: this._options?.environmentConfig?.rotation ?? DefaultViewerOptions.environmentConfig.rotation,
+                visible: this._options?.environmentConfig?.visible ?? DefaultViewerOptions.environmentConfig.visible,
+            };
+            if (this._options?.environmentLighting === this._options?.environmentSkybox) {
+                handleLoadError(this._updateEnvironment(this._options?.environmentLighting, { lighting: true, skybox: true }, this._loadEnvironmentAbortController?.signal));
+            } else {
+                handleLoadError(this._updateEnvironment(this._options?.environmentLighting, { lighting: true }, this._loadEnvironmentAbortController?.signal));
+                handleLoadError(this._updateEnvironment(this._options?.environmentSkybox, { skybox: true }, this._loadSkyboxAbortController?.signal));
+            }
+        }
+
+        if (flags.length === 0 || flags.includes("animation")) {
+            this.animationSpeed = this._options?.animationSpeed ?? DefaultViewerOptions.animationSpeed;
+            this.selectedAnimation = this._options?.selectedAnimation ?? 0;
+            if (this._options?.animationAutoPlay) {
+                this.playAnimation();
+            } else {
+                this.pauseAnimation();
+            }
+        }
+
+        if (flags.length === 0 || flags.includes("camera")) {
+            // In the case of resetting the camera, we always want to restore default states, so convert NaN to undefined.
+            const alpha = Number(this._options?.cameraOrbit?.[0]);

```

Comment:
nit (feel free to ignore this) - we are using `+value` throughout the codebase (smaller footprint). i.e.:
```suggestion
            const alpha = +this._options?.cameraOrbit?.[0];
```

Reply:
Hmm, I gave this a try but I get a TS error saying "Object possibly undefined" (which is true and intentional). Did I miss anything?

Reply:
weird that TS complains about this. `+undefined` is `NaN`, the same as Number(undefined). Keep it this way, I assume Terser is converting it anyhow to `+`